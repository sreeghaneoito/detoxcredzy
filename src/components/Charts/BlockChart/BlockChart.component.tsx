import {View} from 'react-native';
import React from 'react';
import {style} from './BlockChart.style';
import {useTheme} from '../../../hooks/useTheme';
import {BlockChartProps} from '../../../interfaces/component.interface';

/**
 * Block chart
 * @param max: Maximum limt of Chart
 * @param val: Current value of chart
 * @returns JSX Component
 */
const BlockChart = ({max, val}: BlockChartProps) => {
  const {currentTheme} = useTheme();

  // Convert the values to percentage
  const value = ((val || 0) * 100) / max;

  // Set bg based on the percentage
  const activeBg = {
    backgroundColor:
      value <= 40 ? '#E2002B' : value <= 60 ? '#FAC600' : '#28AF51',
  };

  const blockTheme =
    currentTheme === 'dark' ? style.blockdark : style.blocklight;

  try {
    return (
      <View style={style.container}>
        <View
          style={[
            style.block,
            blockTheme,
            style.leftblock,
            value > 0 && activeBg,
            value <= 0 && activeBg,
          ]}
        />
        <View style={[style.block, blockTheme, value >= 20 && activeBg]} />
        <View style={[style.block, blockTheme, value >= 40 && activeBg]} />
        <View style={[style.block, blockTheme, value >= 60 && activeBg]} />
        <View
          style={[
            style.block,
            blockTheme,
            style.rightblock,
            value > 80 && activeBg,
          ]}
        />
      </View>
    );
  } catch (error) {
    console.log('Blockchart crash =>', error);
    return <view style={style.container} />;
  }
};

export default BlockChart;
