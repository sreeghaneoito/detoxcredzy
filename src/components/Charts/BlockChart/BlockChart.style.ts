import {StyleSheet} from 'react-native';

export const style = StyleSheet.create({
  container: {
    height: 30,
    flexDirection: 'row',
  },
  block: {
    height: 30,
    width: '20%',
    borderWidth: 0.5,
  },
  blockdark: {
    borderColor: '#17181A',
    backgroundColor: '#2D2F33',
  },
  blocklight: {
    borderColor: '#FFF',
    backgroundColor: '#F8F8FA',
  },
  rightblock: {
    borderTopRightRadius: 8,
    borderBottomRightRadius: 8,
  },
  leftblock: {
    borderTopLeftRadius: 8,
    borderBottomLeftRadius: 8,
  },
});
