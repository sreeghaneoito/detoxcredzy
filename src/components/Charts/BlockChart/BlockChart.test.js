import React from 'react';
import renderer from 'react-test-renderer';
import BlockChart from './BlockChart.component';

test('renders correctly', () => {
  const tree = renderer.create(<BlockChart />).toJSON();
  expect(tree).toMatchSnapshot();
});
