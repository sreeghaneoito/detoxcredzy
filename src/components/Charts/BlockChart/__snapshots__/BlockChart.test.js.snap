// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "height": 30,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "borderWidth": 0.5,
          "height": 30,
          "width": "20%",
        },
        Object {
          "backgroundColor": "#F8F8FA",
          "borderColor": "#FFF",
        },
        Object {
          "borderBottomLeftRadius": 8,
          "borderTopLeftRadius": 8,
        },
        false,
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "borderWidth": 0.5,
          "height": 30,
          "width": "20%",
        },
        Object {
          "backgroundColor": "#F8F8FA",
          "borderColor": "#FFF",
        },
        false,
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "borderWidth": 0.5,
          "height": 30,
          "width": "20%",
        },
        Object {
          "backgroundColor": "#F8F8FA",
          "borderColor": "#FFF",
        },
        false,
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "borderWidth": 0.5,
          "height": 30,
          "width": "20%",
        },
        Object {
          "backgroundColor": "#F8F8FA",
          "borderColor": "#FFF",
        },
        false,
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "borderWidth": 0.5,
          "height": 30,
          "width": "20%",
        },
        Object {
          "backgroundColor": "#F8F8FA",
          "borderColor": "#FFF",
        },
        Object {
          "borderBottomRightRadius": 8,
          "borderTopRightRadius": 8,
        },
        false,
      ]
    }
  />
</View>
`;
